### DISCUSSION TITLE
Probleme mit CMT2300 und CAN-Bus / Upgrade auf Fusion Pro?

### DISCUSSION THREAD

**Discussion by quantenquark** [hoylabs/OpenDTU-OnBattery#1266] [Q&A]
Hallo zusammen,

ich hoffe auf euer Schwarmwissen, da ich keine Lösung für mein Problem finde.

Ich experimentiere mit dem OpenDTUoB auf einem ESP32-D0WD-V3 (4MB), der auf eine Platine gesteckt ist, auf der ein CMT2300A fest verlötet ist. Den CAN-Bus konnte ich bereits erfolgreich mit einem identischen ESP32 zum Auslesen der Pytes eBox testen.

Nun möchte ich beide Plattformen zusammenbringen. Eine steckbare Lösung zur Verbindung des CAN-Bus mit der ESP/CMT-Platine habe ich gefunden, allerdings bekomme ich im Verbund keine Verbindung mehr zum CMT. Ich habe das Hardwareprofil überprüft, aber komme nicht weiter. Könnte das Problem an der doppelten Nutzung der Pins 14 und 27 für CMT und den CAN-Bus liegen oder habe ich einen ESP der nicht geeignet ist?

Hier ein Auszug aus dem Hardwareprofil:

{
    "name": "CMT, SH1106 Display, Battery",
    "nrf24": {
        "miso": -1,
        "mosi": -1,
        "clk": -1,
        "irq": -1,
        "en": -1,
        "cs": -1
    },
    "cmt": {
        "clk": 12,
        "cs": 27,
        "fcs": 26,
        "sdio": 14
    },
    "display": {
        "type": 3,
        "data": 21,
        "clk": 22
    },
    "battery": {
        "rx": 27,
        "tx": 14
    }
}

Ich löte und bastel eigentlich recht gerne, für meine PV-Anlage wünsche ich mir eine steckerfertige Lösung.
Wenn die 4MB-Version als EOL markiert ist, würde es mehr Sinn machen, direkt auf die Fusion/Fusion Pro zu wechseln?

Vielen Dank im Voraus für eure Hilfe!

Liebe Grüße
Achim

**Comment by Manos1966** [comment]
Hi Achim,

warum "cmt": {"cs": 27, "sdio": 14 und "battery": {"rx": 27 und "tx": 14  ?

Damit wird es garantiert nicht starten koennen
CAN auf andere PINs zuordnen und danach sollte es funktionieren.

und Falls doch nicht, schalte temporaer dein Display aus.

EDIT: und ich war schneler als @spcqike 😜

**Comment by quantenquark** [comment]
Ok, Danke! 
Ich bin jetzt mit RX/TX vom CAN auf Pin 32/33 ausgewichen und es klappt jetzt. 
Die CMTs sind lose ja kaum zu verdrahten, da musste eine fertige Platine her was wiederum einen Engpass bei der Ergänzung des CAN Bus erzeugte. 2-reihige Buchsenleisten sind schon vorteilhaft, mit einer gewinkelten Stiftleiste lies sich der CAN-Bus unterm ESP verdeckt anlöten.

Mit eigenem 3D-Gehäuse und ergänztem RJ45-Port ist das ganze jetzt halbwegs auf der Zielgeraden.

Schön wäre aber eine Platine ohne Bastelaufwand.
Also OpenDTU Fusion mit CAN Erweiterung oder auf die PRO warten?

**Comment by schlimmchen** [comment]
> OpenDTU Fusion mit CAN Erweiterung

Gute Wahl.

> auf die PRO warten?

Gute Wahl, ist aber nicht garantiert, dass du eine kriegst, und wenn dann nicht kurzfristig.
